pyproject.toml

```diff
@@ -281,13 +284,21 @@ extend-exclude = [
     "F401",  # unused import
     "F403",  # star imports
     ]
+"great_expectations/optional_imports.py" = [
+    "TID251",  # flake8-banned-api
+]
 
 [tool.ruff.mccabe]
 max-complexity = 15
 
 [tool.ruff.pydocstyle]
 convention = "google"
 
+[tool.ruff.flake8-tidy-imports]
+[tool.ruff.flake8-tidy-imports.banned-api]
+# Uncomment this to prevent importing sqlalchemy everywhere but optional_imports.py
+# "sqlalchemy".msg = "Don't import sqlalchemy directly, please use the great_expectations.import_manager."
```

Comment:
Commenting in this line will used on this check. Right now we need to fix ~67 more files before we can turn this on for `sqlalchemy`.